### PR TITLE
[FLINK-30237][coordination] Only bundle single ZK client

### PR DESCRIPTION
**Pull request by zentol**
Reverts FLINK-13417 and FLINK-16350 since it's no longer required to bundle 2 different clients.

We'll upgrade to 3.7 in FLINK-29420.

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/test_ha_per_job_cluster_datastream.sh [hunk 1]
```diff
@@ -113,7 +113,6 @@ function run_ha_test() {
     # jm killing loop
     set_config_key "env.pid.dir" "${TEST_DATA_DIR}"
 
-    setup_flink_shaded_zookeeper ${ZOOKEEPER_VERSION}
```
Comment (XComp) [inline flink-end-to-end-tests/test-scripts/test_ha_per_job_cluster_datastream.sh:116]: Is there a reason to keep `${ZOOKEEPER_VERSION}` as a parameter in this method? It's not really needed anymore analogously to the other files where we reverted mentioning the ZK version in the log output.

Comment (XComp) [reply flink-end-to-end-tests/test-scripts/test_ha_per_job_cluster_datastream.sh:116]: We're even removing the ZK version in `run-nightly-tests.sh` as a parameter of `test_ha_per_job_cluster_datastream.sh`. So, I guess, we can clean it up entirely here.

**Review by XComp — CHANGES_REQUESTED** [review]
LGTM in general. Only one missing cleanup should be done. See my comment below. I verified this diff based on the PRs #11173, #11288 and #12504

File: flink-end-to-end-tests/test-scripts/test_ha_per_job_cluster_datastream.sh (hunk 1 above)
Comment (zentol) [reply flink-end-to-end-tests/test-scripts/test_ha_per_job_cluster_datastream.sh:116]: No I just missed it.

**Review by XComp — APPROVED** [review]
LGTM :+1:

**Pull request merged**